src/app/view/components/dot-navigation/components/dot-nav-item/dot-nav-item.component.ts

```diff
@@ -28,5 +57,55 @@ export class DotNavItemComponent {
             originalEvent: $event,
             data: data
         });
+        this.dotEventsService.notify('hide-sub-nav-fly-out');
+    }
+
+    /**
+     * Handle right-click on menu section title
+     *
+     * @param MouseEvent $event
+     * @memberof DotNavItemComponent
+     */
+    @HostListener('contextmenu', ['$event'])
+    showSubMenuPanel(event: MouseEvent) {
+        if (this.collapsed) {
+            event.preventDefault();
+            this.dotEventsService.notify('hide-sub-nav-fly-out');
+            this.iframeOverlayService.show();
+            this.contextmenu = true;
+        }
+    }
+
+    /**
+     * Handle click on document to hide the fly-out menu
+     *
+     * @memberof DotNavItemComponent
+     */
+    @HostListener('document:click')
+    handleDocumentClick(): void {
+        this.contextmenu = false;
+    }
+
+    /**
+     * Handle click in dot-sub-nav items
+     *
+     * @param { originalEvent: MouseEvent; data: DotMenuItem } $event
+     * @memberof DotNavItemComponent
+     */
+    handleItemClick(event: { originalEvent: MouseEvent; data: DotMenuItem }) {
+        this.itemClick.emit(event);
+        this.dotEventsService.notify('hide-sub-nav-fly-out');
+    }
+
+    private setHideFlyOutSubscription(): void {
```

Comment:
I think this method's name not should have the set prefix because I expect that a method with set... receive a params to be set